docs/website/docs/deployment-configurations/bare-metal.md

```diff
@@ -0,0 +1,108 @@
+# Run on a Bare-Metal Platform
+
+IREE supports CPU model execution on a bare-metal platform. That is, a platform
+without operating system support and the executable runs on the machine mode.
+The IREE compiler follows the path to generate the code runs on the
+[dynamic library CPU HAL driver](https://google.github.io/iree/deployment-configurations/cpu-dylib/)
+and a subset of runtime library can be used to execute the model.
+
+## Prerequisites
+
+Out-of-tree bare-metal platform tools and source code for the system should be
+ready, such as
+
+* Compilation toolchain
+* Platform linker script
+* Firmware libraries
+
+Please follow the
+[instructions](https://google.github.io/iree/deployment-configurations/cpu-dylib/#get-compiler-for-cpu-native-instructions)
+to retrieve the IREE compiler.
+
+## Compile the model for bare-metal
+
+The model can be compiled with the following command from the IREE compiler
+build directory
+
+``` shell hl_lines="3 4 5 6"
+iree/tools/iree-translate \
+    -iree-mlir-to-vm-bytecode-module \
+    -iree-hal-target-backends=dylib-llvm-aot \
+    -iree-llvm-link-embedded=true \
+    -iree-llvm-target-triple=x86_64-pc-linux-elf \
+    -iree-llvm-debug-symbols=false \
+    iree/samples/models/simple_abs.mlir \
+    -o /tmp/simple_abs_dylib.vmfb
+
+```
+
+In which
+
+* `iree-hal-target-backends=dylib-llvm-aot`: Build the model for the dynamic
+library CPU HAL driver
+* `iree-llvm-link-embedded=true`: Generate the dynamic library with
+[LLD](https://lld.llvm.org/) and the artifact can be loaded with the
+[embedded library loader](https://github.com/google/iree/blob/main/iree/hal/local/loaders/embedded_library_loader.h)
+without invoking the dynamic library support
+* `iree-llvm-target-triple`: Use the `<arch>-pc-linux-elf` LLVM target triple so
+the artifact has a fixed ABI to be rendered by the
+[elf_module library](https://github.com/google/iree/tree/main/iree/hal/local/elf)
+* `iree-llvm-debug-symbols=false`: To reduce the artifact size
+
+See [generate.sh](https://github.com/google/iree/blob/main/iree/hal/local/elf/testdata/generate.sh)
+for example command-line instructions of some common architectures
+
+You can replace the MLIR file with the other MLIR model files, following the
+[instructions](https://google.github.io/iree/deployment-configurations/cpu-dylib/#compile-the-model)
+
+!!! todo
+    add the static library support
+
+## Build bare-metal runtime from the source (CMake only)
```

Comment:
I'd just drop "(Cmake only)" from here. I've intentionally omitted all mention of Bazel from the website :P

```suggestion
## Build bare-metal runtime from the source
```

Reply:
Done.